src/datahike/query.cljc

```diff
@@ -539,11 +542,44 @@
      :clj  (when (namespace sym)
              (when-some [v (resolve sym)] @v))))
 
+(def ^:private find-method
+  #?(:cljs nil
+     :clj (memoize
+           (fn find-method-impl [^Class this-class method-name args-classes]
+             (or (->> this-class
+                      .getMethods
+                      (some (fn [^Method method]
+                              (when (and (= method-name (.getName method))
+                                         (= (count args-classes)
+                                            (.getParameterCount method))
+                                         (every? true? (map #(Reflector/paramArgTypeMatch %1 %2)
+                                                            (.getParameterTypes method)
+                                                            args-classes)))
+                                method))))
+                 (throw (ex-info (str (.getName this-class) "."
+                                      method-name "("
+                                      (str/join "," (map #(.getName ^Class %) args-classes))
+                                      ") not found")
+                                 {:this-class this-class
+                                  :method-name method-name
+                                  :args-classes args-classes})))))))
```

Comment:
Note: I memoize the results of the method search to improve performance. 
Note: I use `.getMethods` with manual parameter arg checking instead of `.getMethod` because the latter returns result for exact parameter argument match instead of assignable match. For example, `(.getMethod String "contains" (into-array Class [String]))` can't find contains method because it expects `CharSequence` and not a `String` as input argument even though strings are char sequences.